internal/provider/resource_mackerel_channel.go

```diff
@@ -0,0 +1,255 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mackerelio/mackerel-client-go"
+)
+
+func resourceMackerelChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMackerelChannelCreate,
+		Read:   resourceMackerelChannelRead,
+		Update: resourceMackerelChannelUpdate,
+		Delete: resourceMackerelChannelDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"emails": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			// Field name may only contain lowercase alphanumeric characters & underscores.
+			"user_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"events": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"mentions": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"enabled_graph_image": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceMackerelChannelCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*mackerel.Client)
+
+	input, err := buildChannelParameter(d)
+	if err != nil {
+		return err
+	}
+
+	channel, err := client.CreateChannel(input)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] mackerel channel %q created.", channel.ID)
+	d.SetId(channel.ID)
+
+	return resourceMackerelChannelRead(d, meta)
+}
+
+func resourceMackerelChannelRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*mackerel.Client)
+
+	log.Printf("[DEBUG] Reading mackerel channel: %q", d.Id())
+	channels, err := client.FindChannels()
+	if err != nil {
+		return err
+	}
+
+	for _, channel := range channels {
+		if channel.ID == d.Id() {
+			_ = d.Set("id", channel.ID)
+			_ = d.Set("name", channel.Name)
+			_ = d.Set("type", channel.Type)
+			_ = d.Set("url", channel.URL)
+			_ = d.Set("enabledGraphImage", channel.EnabledGraphImage)
+			_ = d.Set("user_ids", channel.UserIDs)
+			_ = d.Set("mentions", channel.Mentions)
+			_ = d.Set("events", channel.Events)
+			break
+		}
+	}
+
+	return nil
+}
+
+// DeleteChannel() -> CreateChannel()
+// Bacause, Mackerel API does not have UpdateChannel()
```

Comment:
I think it's better to use [ForceNew](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#forcenew) to make it clear to the user that the `lD` of the resource on Mackerel will change.

Reply:
Could you give me an idea?
I added `ForceNew: true` for all resources, and I got following error message from terraform. Or you don't mean this?
```
1 error occurred:
        * resource mackerel_channel: All fields are ForceNew or Computed w/out Optional, Update is superfluous
```

Reply:
Can you try this like?
https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/resource_aws_config_configuration_aggregator.go#L28-L37

Reply:
I'll try that, also I found the following approach.
In this, they remove `Update` function, and it works.
Anyway, I'll implement yours.

https://github.com/DataDog/terraform-provider-datadog/blob/90663e3f36a057cf9d0c588c21f9cee144a40631/datadog/resource_datadog_integration_aws_lambda_arn.go#L27-L50

Reply:
Thanks information.
I think looks like good that resource_datadog_integration_aws_lambda_arn approach.

It doesn't matter which one you implement.

Reply:
I've fixed this with DataDog approach. please check it.